presto-orc/src/test/java/com/facebook/presto/orc/AbstractDwrfStripeCachingTest.java

```diff
@@ -0,0 +1,286 @@
+    }
+
+    static File getResourceFile(String fileName)
+    {
+        String resourceName = "dwrf_stripe_cache/" + fileName;
+        return new File(getResource(resourceName).getFile());
+    }
+
+    private static Object[][] toArray(File file1, File file2)
+    {
+        return new Object[][] {{file1}, {file2}};
+    }
+
+    /**
+     * Create an ORC file in a way that the stripe cache would only have data for
+     * half of the stripes.
+     */
+    private static TempFile rewriteFile(TempFile srcFile, DwrfStripeCacheMode mode)
+    {
+        DwrfProto.Footer footer = readFileFooter(srcFile.getFile());
+        assertEquals(footer.getStripesCount(), 4);
+
+        // calculate stripe cache size to fit data exactly for half (2) stripes
+        long size = 0;
+        for (int i = 0; i < 2; i++) {
+            DwrfProto.StripeInformation stripe = footer.getStripes(i);
+            if (mode.hasFooter()) {
+                size += stripe.getFooterLength();
+            }
+            if (mode.hasIndex()) {
+                size += stripe.getIndexLength();
+            }
+        }
+
+        // now create the same file but with the specifically configured cache
+        // size to fit data only for two stripes
+        return writeOrcFile(true, mode, new DataSize(size, BYTE));
+    }
+
+    /**
+     * Creates a file with 3 INT columns and 4 stripes with 100 rows each with the
+     * following values:
+     * Column 0: row number
+     * Column 1: Integer.MAX_VALUE
+     * Column 2: row number * 10
+     */
+    private static TempFile writeOrcFile(boolean cacheEnabled, DwrfStripeCacheMode cacheMode, DataSize cacheMaxSize)
+    {
+        TempFile outputFile = new TempFile();
+        try {
+            Type type = INTEGER;
+            List<Type> types = ImmutableList.of(type, type, type);
+            OrcWriterOptions writerOptions = OrcWriterOptions.builder()
+                    .withStripeMaxRowCount(100)
+                    .build();
+            DwrfWriterOptions dwrfWriterOptions = DwrfWriterOptions.builder()
+                    .withStripeCacheEnabled(cacheEnabled)
+                    .withStripeCacheMode(cacheMode)
+                    .withStripeCacheMaxSize(cacheMaxSize)
+                    .build();
+
+            OrcWriter writer = new OrcWriter(
+                    new OutputStreamDataSink(new FileOutputStream(outputFile.getFile())),
+                    ImmutableList.of("Int1", "Int2", "Int3"),
+                    types,
+                    DWRF,
+                    ZLIB,
+                    Optional.empty(),
+                    NO_ENCRYPTION,
+                    writerOptions,
+                    Optional.of(dwrfWriterOptions),
+                    ImmutableMap.of(),
+                    HIVE_STORAGE_TIME_ZONE,
+                    true,
+                    BOTH,
+                    new OrcWriterStats());
+
+            // write 4 stripes with 100 values each
+            int cnt = 0;
```

Comment:
nit: count. Let's not use abbreviation :)

Reply:
fixed